src/server.rs

```diff
@@ -1497,42 +1493,9 @@ where
     #[inline]
     fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
         match *self {
-            Handshaking::Flushing(_) => write!(f, "Handshaking::Flushing(_)"),
-            Handshaking::ReadingPreface(_) => write!(f, "Handshaking::ReadingPreface(_)"),
-            Handshaking::Empty => write!(f, "Handshaking::Empty"),
+            Handshaking::Flushing(_) => write!(f, "Flushing(_)"),
+            Handshaking::ReadingPreface(_) => write!(f, "ReadingPreface(_)"),
+            Handshaking::Done => write!(f, "Done"),
```

Comment:
minor nit: these could probably be changed to 
```suggestion
            Handshaking::Flushing(_) => f.write_str("Flushing(_)"),
            Handshaking::ReadingPreface(_) =>  f.write_str("ReadingPreface(_)"),
            Handshaking::Done =>  f.write_str("Done"),
```
the `write!` macro is unnecessary, and if we just used `write_str`, we could avoid some macro expansions, which might have a tiny effect on build time or something. but, it also doesn't really matter either way --- just maybe worth doing while we're changing this code.

I dunno why I ever used `write!` here --- probably, I just didn't know better at the time :)